homeassistant/components/switchbot_cloud/cover.py

```diff
@@ -0,0 +1,233 @@
+"""Support for the Switchbot Bot as a Cover."""
+
+import asyncio
+from typing import Any
+
+from switchbot_api import (
+    BlindTiltCommands,
+    CommonCommands,
+    CurtainCommands,
+    Device,
+    Remote,
+    RollerShadeCommands,
+    SwitchBotAPI,
+)
+
+from homeassistant.components.cover import (
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import SwitchbotCloudData, SwitchBotCoordinator
+from .const import DOMAIN
+from .entity import SwitchBotCloudEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up SwitchBot Cloud entry."""
+    data: SwitchbotCloudData = hass.data[DOMAIN][config.entry_id]
+    async_add_entities(
+        _async_make_entity(data.api, device, coordinator)
+        for device, coordinator in data.devices.covers
+    )
+
+
+class SwitchBotCloudCover(SwitchBotCloudEntity, CoverEntity):
+    """Representation of a SwitchBot Cover."""
+
+    _attr_name = None
+    _attr_is_closed: bool | None = None
+
+    def _set_attributes(self) -> None:
+        if self.coordinator.data is None:
+            return
+        position: int | None = self.coordinator.data.get("slidePosition")
+        if position is None:
+            return
+        self._attr_current_cover_position = 100 - position
+        self._attr_current_cover_tilt_position = 100 - position
+        self._attr_is_closed = position == 100
+
+
+class SwitchBotCloudCoverCurtain(SwitchBotCloudCover):
+    """Representation of a SwitchBot Curtain & Curtain3."""
+
+    _attr_device_class = CoverDeviceClass.CURTAIN
+    _attr_supported_features: CoverEntityFeature = (
+        CoverEntityFeature.OPEN
+        | CoverEntityFeature.CLOSE
+        | CoverEntityFeature.STOP
+        | CoverEntityFeature.SET_POSITION
+    )
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the cover."""
+        await self.send_api_command(CommonCommands.ON)
+        await asyncio.sleep(10)
+        await self.coordinator.async_request_refresh()
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the cover."""
+        await self.send_api_command(CommonCommands.OFF)
+        await asyncio.sleep(10)
+        await self.coordinator.async_request_refresh()
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Move the cover to a specific position."""
+        position: int | None = kwargs.get("position")
+        if position is not None:
+            await self.send_api_command(
+                CurtainCommands.SET_POSITION,
+                parameters=f"{0},ff,{100 - position}",
```

Comment:
what's the ff for?

Reply:
it mean "default mode", 
<img width="1407" height="264" alt="image" src="https://github.com/user-attachments/assets/aff5e403-3583-4684-824c-e21db05b039d" />
https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#curtain-2